src/MotionHandler.moon

```diff
@@ -171,21 +171,55 @@ class MotionHandler
 				else
 					newText = line\detokenizeTransformsCopy timeDelta
 
-				newText = newText\gsub "\\fade(%b())", ( fade ) ->
-					a1, a2, a3, t1, t2, t3, t4 = fade\match("(%d+),(%d+),(%d+),(%d+),(%d+),(%d+),(%d+)")
-					if t1 == nil
-						message = "There is a malformed \\fade you must fix.\n\\fade requires 7 integer arguments.\nLine: #{.number}, tag: \\fade(#{fade})."
-						if fade\match("(%d+),(%d+)")
-							message ..= "\nPerhaps you meant to use \\fad."
-						log.windowError message
-
-					t1, t2, t3, t4 = tonumber( t1 ), tonumber( t2 ), tonumber( t3 ), tonumber( t4 )
-					-- beautiful.
-					t1 -= timeDelta
-					t2 -= timeDelta
-					t3 -= timeDelta
-					t4 -= timeDelta
-					("\\fade(%s,%s,%s,%d,%d,%d,%d)")\format a1, a2, a3, t1, t2, t3, t4
+				if @options.main.killTrans
+				-- multiplies every alpha tag by a scaling factor determined by the fade envelope
+
+					-- if there's a better way to do this I would love to know
+					a1, a2, a3, t1, t2, t3, t4 = 0, 0, 0, 0, 0, 0, 0
+					line\runCallbackOnFirstOverride ( tagBlock ) =>
```

Comment:
I don't think there's any guarantee that `\fade` is in the first override block, though the tag de-duplication should at least make sure there is only one in the whole line. This should run on all overrides.